connector/datadogconnector/connector.go

```diff
@@ -62,7 +63,7 @@ func newConnector(set component.TelemetrySettings, _ component.Config, metricsCo
 	ctx := context.Background()
 	return &connectorImp{
 		logger:          set.Logger,
-		agent:           datadog.NewAgent(ctx, in),
+		agent:           newTraceAgent(ctx, cfg.(*Config), in),
```

Comment:
Can we make changes to `datadog` package, rather than create new Trace Agent here ?

Reply:
Same, prefer not to do it at least in this PR.

Reply:
datadog package has some logic in both `Start` and `Stop` . We should try to make changes in datadog package. It only requires on one method.

Reply:
Done https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30787/commits/b472520ab9a215e630f2f8a108fc4df817d2697f